### PR TITLE
feat: Adding the possibility of copying the link by clicking text and icons

### PR DESCRIPTION
**Pull request by pierrbt**
I added the possibility to copy the link just by clicking the text on the upload dialog, and I also added a small check icon when the copying is done. 

Tell me if you think something can be changed. I'll be happy to help for other things on this project, I've already used quite a lot TypeScript with React and also Vanilla NodeJS.

### CODE REVIEW

File: frontend/src/components/upload/modals/showCompletedUploadModal.tsx [hunk 1]
```diff
@@ -27,22 +28,34 @@ const Body = ({ share, appUrl }: { share: Share; appUrl: string }) => {
...
           window.isSecureContext && (
-            <ActionIcon
-              onClick={() => {
-                clipboard.copy(link);
-                toast.success("Your link was copied to the keyboard.");
-              }}
-            >
-              <TbCopy />
+            <ActionIcon onClick={copyLink}>
+              {checkState ? <TbChecks /> : <TbCopy />}
```
Comment (stonith404) [inline frontend/src/components/upload/modals/showCompletedUploadModal.tsx:58]:
```suggestion
              {checkState ? <TbCheck /> : <TbCopy />}
```
I think the icon with the single check mark would be a bit cleaner.

**Review by stonith404 — COMMENTED** [review]
Thank you, this is a nice addition! 

In the file `frontend/src/components/share/modals/showCompletedReverseShareModal.tsx` is the exact same component, it would be nice if you can change it there too because else the design is inconsistent. 

The best way would be if your create a new component e.g `CopyTextField` and use it in `showCompletedReverseShareModal.tsx` and `showCreateUploadModal.tsx`.

File: frontend/src/components/upload/modals/showCompletedUploadModal.tsx (hunk 1 above)
Comment (pierrbt) [reply frontend/src/components/upload/modals/showCompletedUploadModal.tsx:58]: Ok, I'll create a new component and add it in the other file asap

**Comment by pierrbt** [comment]
@stonith404, I've made the requested changes, let me know what you think about it

**Comment by stonith404** [comment]
Awesome, thank you! Just one thing; when I want to copy the link again, by pressing on the `CopyTextField`, it doesn't copy the link again. Is this on purpose or a bug?

**Comment by pierrbt** [comment]
> when I want to copy the link again, by pressing on the CopyTextField, it doesn't copy the link again. Is this on purpose or a bug?

Yes, you can only copy the text itself once because if the user want to select it anyway, it would notificate 3x or more, so it selects only once on the text, but he can always click on the clipboard if he wants to copy it again.

I found disturbing that it displays multiple notifications if we only wanted to select the link. 

I can change it if you want, it's only two lines : 

```tsx
const [textClicked, setTextClicked] = useState(false);

if (!textClicked) {
    copyLink();
    setTextClicked(true);
}
```

**Comment by pierrbt** [comment]
@stonith404

**Comment by stonith404** [comment]
Makes absolutely sense!

**Review by stonith404 — APPROVED** [review]

**Comment by stonith404** [comment]
Are you able to merge the PR or do I have to do it?

**Comment by pierrbt** [comment]
No I don't have write access, even with a review, I can't.

**Pull request merged**